src/test/scala/com/amazon/deequ/utils/FixtureSupport.scala

```diff
@@ -202,21 +205,53 @@ trait FixtureSupport {
       (1, 4),
       (2, 5),
       (3, 6)
-    ).toDF("att1", "att2")
+    ).toDF("]att1[", "att2")
   }
 
   def getDfWithCategoricalColumn(
       sparkSession: SparkSession,
       numberOfRows: Int,
       categories: Seq[String])
-    : DataFrame = {
+  : DataFrame = {
 
     val random = new Random(0)
 
     import sparkSession.implicits._
     (1 to numberOfRows)
       .toList
       .map { index => (s"$index", random.shuffle(categories).head)}
-      .toDF("att1", "categoricalColumn")
+      .toDF("]att1[", "categoricalColumn")
+  }
+
+  implicit object OrderingTestMap extends Ordering[Map[String, Any]] {
```

Comment:
can you explain why we need this?

Reply:
I added this comment:

(tl;dr to make the JSON deserialize tests ignore ordering since most of the tests don't need to explicitly check for ordering)

```
A map ordering to ensure that JSON array deserialization results are
checked in a consistent manner.

In general, an `Ordering[Map[String,Any]]` is not a great pattern: here,
however, we have a need to check that various JSON deserialization-based
tests correctly parse & handle arrays of objects. Instead of ensuring
that the ordering inherint in the element processing of the various bits
of code under test do not permute their outputs (something that _is not_
a part of any of the code's stated contracts), the idea here is to re-use
_this_ ordering definition.

This JSON map ordering definiton is specific to the deequ tests: the calls
insde the over-loaded `compare` method have a specific, deterministic
order of hard-coded keys to check when determining which of two JSON
deserialized objects is before another in a sorted collection.
```